src/server/actions/saveChapter.js

```diff
@@ -1,14 +1,25 @@
 import getChapter from 'src/server/actions/getChapter'
 import {Chapter} from 'src/server/services/dataService'
+import {LGBadRequestError} from 'src/server/util/error'
+
+export default async function saveChapter(chapter) {
+  const {id} = chapter || {}
+  chapter.name = chapter.name.toLowerCase().replace(/\s/g, '')
+
+  const chaptersWithSameName = await Chapter.filter(row => {
+    return row('name').downcase().eq(chapter.name)
+  })
+
+  if (chaptersWithSameName.length > 0) {
+    throw new LGBadRequestError('Chapter name must be unique')
+  }
```

Comment:
Also, this function is used for both insert and update, so there's some special consideration we need to make when applying this check. It's possible that, for example, the input values contain the name, but it also includes an ID and is an update for an existing chapter. In that case, this might reject the change when it shouldn't.